operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -49,11 +54,26 @@ public KubernetesDependentResource(Class<R> resourceType, String name) {
     super(resourceType, name);
   }
 
+  public KubernetesDependentResource(
+      Class<R> resourceType, String name, SSABasedGenericKubernetesResourceMatcher<R> matcher) {
+    this(resourceType, name);
+    this.matcher = matcher;
+    matcherSet = true;
+  }
+
   @Override
   public void configureWith(KubernetesDependentResourceConfig<R> config) {
     this.kubernetesDependentResourceConfig = config;
   }
 
+  public void setMatcher(SSABasedGenericKubernetesResourceMatcher<R> matcher) {
+    if (matcherSet) {
```

Comment:
Why is this explicitly forbidden?

Reply:
Because it doesn't make sense to change the matching strategy after the resource has been used already. If anything, it should probably be set when the dependent is created (which is an argument to put it on the config only).